addons/mail/static/src/models/emoji_registry.js

```diff
@@ -0,0 +1,98 @@
+/** @odoo-module **/
+
+import { registerModel } from '@mail/model/model_core';
+import { many } from '@mail/model/model_field';
+import { insertAndReplace } from '@mail/model/model_field_command';
+
+registerModel({
+    name: 'EmojiRegistry',
+    identifyingFields: ['messaging'],
+    fields: {
+        allEmojis: many('Emoji', {
+            default: insertAndReplace([
+                { unicode: "😊", sources: [":)", ":-)", "=)", ":]"], description: ":)" },
+                { unicode: "😃", sources: [":D", ":-D", "=D"], description: ":D" },
+                { unicode: "😆", sources: ["xD", "XD"], description: "xD" },
```

Comment:
I find it interesting to point out that the very same information could be represented using _Unicode code point escapes_:
```suggestion
                { unicode: "\u{1F60A}", sources: [":)", ":-)", "=)", ":]"], description: ":)" },
                { unicode: "\u{1F603}", sources: [":D", ":-D", "=D"], description: ":D" },
                { unicode: "\u{1F606}", sources: ["xD", "XD"], description: "xD" },
```
I'm not saying you shall use it instead of the bare characters, though. Both approaches seem to have their own tradeoffs; I just wanted to highlight this other possibility so that it can also be considered.

Reply:
Thank you for the advice. Storing the information as Unicode code point escapes would make more sense (seeing emojis in code is a bit jarring, and maybe some code editor wouldn't display emojis properly in code)

However, it would hurt readability a lot, so it may be good to put the displayed emoji as a comment at the end of each line.

I will discuss it with my team, to see what could be the best approach :)